examples/ex10p.cpp

```diff
@@ -646,10 +646,7 @@ real_t HyperelasticOperator::ElasticEnergy(const ParGridFunction &x) const
 
 real_t HyperelasticOperator::KineticEnergy(const ParGridFunction &v) const
 {
-   real_t loc_energy = 0.5*M.InnerProduct(v, v);
-   real_t energy;
-   MPI_Allreduce(&loc_energy, &energy, 1, MPITypeMap<real_t>::mpi_type,
-                 MPI_SUM, fespace.GetComm());
+   real_t energy = 0.5*M.InnerProduct(v, v);
```

Comment:
This kind of change will break user codes since the meaning of `InnerProduct` becomes overloaded and it will call the version that does the global reduction while before there was no global reduction.

A simple solution is to use a different name for the new method, e.g. `ParInnerProduct`.

This same observation affects the change in `examples/sundials/ex10p.cpp` and in `miniapps/electromagnetics/joule_solver.cpp`.

Reply:
It is not exactly overloaded, because it takes `ParGridFuction`s as arguments, but it is true it may break something. It is the kind of thing that makes sense from the design point of view, but it was better to do in the early days and it is hard to go back now 😕 . Ok, will change it... 😉